src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -1017,6 +1029,20 @@ public SIPDialog getDialog(String dialogId) {
 
     }
 
+    private SIPDialog findSIPDialogFromInjectionListeners(String dialogId) {
+        SIPDialog dialog = null;
+        synchronized (dialogInjectionListeners) {
```

Comment:
The add/remove Listener should be sync too, so we should sync here aswell to ensure that what we won't to read is still available